crates/obj/src/context.rs

```diff
@@ -22,20 +19,12 @@ pub fn layout_vmcontext(
     let out_len = ofs.size_of_vmctx() as usize;
     let mut out = vec![0; out_len];
 
-    // Assign unique indicies to unique signatures.
-    let mut signature_registry = HashMap::new();
-    let mut signature_registry_len = signature_registry.len();
-    for (index, sig) in module.signatures.iter() {
-        let offset = ofs.vmctx_vmshared_signature_id(index) as usize;
-        let target_index = match signature_registry.entry(sig) {
-            Entry::Occupied(o) => *o.get(),
-            Entry::Vacant(v) => {
-                assert_le!(signature_registry_len, std::u32::MAX as usize);
-                let id = TargetSharedSignatureIndex::new(signature_registry_len as u32);
-                signature_registry_len += 1;
-                *v.insert(id)
-            }
-        };
+    for (offset, target_index) in module.signatures.iter().map(|(index, _)| {
```

Comment:
Previously, we were deduplicating our signatures here. Now that this is done with `ModuleEnvironment`, this code is simplified a little.